doc/advanced_configuration.rst

```diff
@@ -520,20 +522,118 @@ The highest precedence is always given to the `-D` flag of the
 ``sphinx-build`` command.
 
 
-.. _find_mayavi:
+.. _image_scrapers:
 
-Finding Mayavi figures
-======================
-By default, Sphinx-gallery will only look for :mod:`matplotlib.pyplot` figures
-when building. However, extracting figures generated by :mod:`mayavi.mlab` is
-also supported. To enable this feature, you can do::
+Image scrapers
+==============
+
+By default, at the end of each code block Sphinx-gallery will detect new
+:mod:`matplotlib.pyplot` figures and turn them into images. This behavior is
+equivalent to the default of::
+
+    sphinx_gallery_conf = {
+        ...
+        'image_scrapers': ('matplotlib',),
+    }
+
+Built-in support is also provided for finding :mod:`Mayavi <mayavi.mlab>`
+figures. To enable this feature, you can do::
+
+    sphinx_gallery_conf = {
+        ...
+        'image_scrapers': ('matplotlib', 'mayavi'),
+    }
+
+.. note:: The parameter ``find_mayavi_figures`` which can also be used to
+          extract Mayavi figures is **deprecated** in version 1.13+,
+          and will be removed in a future release.
+
+You can also add your own custom function (or callable class instance)
+to this list. See :func:`sphinx_gallery.gen_rst.matplotlib_scraper` for
+a description of the interface. Here is pseudocode for what this function
+and :func:`sphinx_gallery.gen_rst.mayavi_scraper` do under the hood, which
+uses :func:`sphinx_gallery.gen_rst.figure_rst` to create standardized RST::
+
+    def mod_scraper(image_path_iterator, gallery_conf)
+        import mymod
+        image_names = list()
+        for fig, image_path in zip(mymod.get_figures(), image_path_iterator):
+            fig.save_png(image_path)
+            image_names.append(image_path)
+        mymod.close('all')
+        return figure_rst(image_names, gallery_conf['src_dir'])
+
+
+For example, a naive class to scrape any new PNG outputs in the
+current directory could do, e.g.::
+
+    import glob
+    import shutil
+    from sphinx_gallery.gen_rst import figure_rst
+
+
+    class PNGScraper(object):
+        def __init__(self):
+            self.seen = set()
 
+        def __call__(self, image_path_iterator, gallery_conf):
+            count = 0
+            pngs = sorted(glob.glob(os.path.join(os.getcwd(), '*.png'))
+            image_names = list()
+            for png in my_pngs:
+                if png not in seen:
+                    seen |= set(png)
+                    image_names.append(image_path_iterator.next())
+                    shutil.copyfile(png, image_names[-1])
+            return figure_rst(image_names, gallery_conf['src_dir'])
+
+
+    sphinx_gallery_conf = {
+        ...
+        'image_scrapers': ('matplotlib', PNGScraper()),
+    }
+
+If you have an idea for a scraper that is likely to be useful to others
+(e.g., a fully functional image-file-detector, or one for another visualization
+libarry), feel free to contribute it on GitHub!
+
+
+.. _reset_modules:
+
+Resetting modules
```

Comment:
As @lesteve always reminds me. We should have different features in different PRs

Reply:
I see these as a bit related, though, since if you don't want to use the `mpl` scraper then you probably also do not want to use the `mpl` resetter.

Reply:
(And if you want to add your own scraper, you might need your own resetter.)